### ISSUE TITLE
Introduction of automations (GitHub Actions)

### ISSUE THREAD

**Issue by dimsanius** [Meru3m/virtualbox-snapshotter#17]
Since we already have linter in place, it would be a good idea to introduce some automations:

- Implement automation to run linter
- Implement automation to run unit tests

For this, we would need:
- [x] Your consent to have GitHub Actions implemented on your repo
- [x] Finalise linter rules
  - [x] Implement automation to run linter on PR to `master` and on push to `master`
- [ ] Implement Unit Tests
  - [ ] Implement automation to run unit test on PR to `master` and on push to `master`

**Comment by mattiabaldari** [comment]
Yeah, no problem to activate github actions, I am just not used to it.
I must read something to understand it. I'll let you know ;)
